_benchmark/workloads/vectorsearch.md

```diff
@@ -0,0 +1,569 @@
+| `query_body`                              | The JSON properties that will be merged with search body                                                                                        |
+| `search_clients`                          | The number of clients to use for running queries.                                                                                               |
+| `repetitions`                             | THe number of repetitions until the data set is exhausted. Default is `1`.                                                                      |
+| `target_throughput`                       | The target throughput for each query operation in requests per second. Default is `10`.                                                         |
+| `time_period`                             | The period of time dedicated for the run the benchmark test in seconds. Default is `900`.                                                       |
+
+
+
+## Test procedures
+
+The workload supports the following test procedures.
+
+### No Train Test
+
+The No Train Test procedure tests vector search indices that require no training. You can define the underlying configuration of the vector search algorithm (such as specific engine, space type, etc.) as method definitions. Check vector search method definitions for more details.
+
+### No Train Test Index Only
+
+Th No Train Test Index Only procedure is used to index vector search indices that require no training. It's particularly useful when you want to benchmark only the indexing operation.
+
+### Force Merge Index
+
+The Force Merge Index procedure optimizes vector search indices by performing force merge operations up to a given maximum number of segments. For large datasets, force merge is a costly operation. Therefore, it's better to have a separate procedure to trigger force merge operations occasionally based on user requirements.
+
+### Search
+
+The Search procedure benchmarks previously indexed vector search indices. It's useful when you want to benchmark large vector search indices without re-indexing each time, since load time can be substantial for large datasets. This procedure includes warmup operations to avoid cold start problems during vector search.
+
+### No Train Test AOSS
+
+The No Train Test AOSS procedure is similar to the No Train Test procedure but is specifically targeted for Amazon OpenSearch Serverless Vector Search Collection. This procedure doesn't include operations like **refresh** and **warmup** since they aren't supported by Vector Search Collection.
+
+### Train Test
+
+The Train Test procedure benchmarks approximate k-NN search algorithms that require a training step. For example, the FAISS IVF requires a training step to cluster vectors. After the step is performed, the benchmark can search against a smaller number of cluster centroids instead of the entire dataset.
```

Comment:
```suggestion
The Train Test procedure benchmarks approximate k-NN search algorithms that require a training step. For example, the Faiss IVF requires a training step to cluster vectors. After the step is performed, the benchmark can search against a smaller number of cluster centroids instead of the entire dataset.
```